src/memory/primitives/relocatable.ts

```diff
@@ -0,0 +1,80 @@
+import { Felt } from './felt';
+import { Uint, UnsignedInteger } from './uint';
+
+export class RelocatableError extends Error {}
+export class OffsetUnderflow extends RelocatableError {}
+export class OffsetOverflow extends RelocatableError {}
+export class SegmentError extends RelocatableError {}
+export class ForbiddenOperation extends RelocatableError {}
+export class TypeError extends RelocatableError {}
+export class InternalError extends RelocatableError {}
+
+export type MaybeRelocatable = Relocatable | Felt;
```

Comment:
The structure is a little weird no? Because you implement some operations between Relocatables and MaybeRelocatables. I would keep it the cairo-vm-go way:
addition with `add` is only allowed between the same types. If you want to add a Felt to a MaybeRelocatable for example, I would explicitly have a `addFelt` function.

I also think we should maybe just implement addition between types, and `addFelt` (I think this is what lambda class did).